python/ray/tests/test_basic_2.py

```diff
@@ -720,6 +720,46 @@ def get(self):
         assert b"OK" in output
 
 
+@pytest.mark.skipif(client_test_enabled(), reason="init issue")
+def test_use_dynamic_function_and_class():
+    # Test use dynamically defined functions
+    # and classes for remote tasks and actors.
+    # See https://github.com/ray-project/ray/issues/12834.
+
+    current_path = os.path.dirname(__file__)
+    job_config = ray.job_config.JobConfig(code_search_path=[current_path])
+    ray.init(job_config=job_config)
+
+    def foo1():
+        @ray.remote
+        def foo2():
+            return "OK"
+
+        return foo2
+
+    @ray.remote
+    class Foo:
+        @ray.method(num_returns=1)
+        def foo(self):
+            return "OK"
+
+    f = foo1()
+    assert ray.get(f.remote()) == "OK"
+    key_ = (
+        b"RemoteFunction:" + ray.worker.global_worker.current_job_id.binary() +
+        b":" + f._function_descriptor.function_id.binary())
+    (job_id_str, function_id_str, function_name, serialized_function, module,
+     max_calls) = ray.worker.global_worker.redis_client.hmget(
+         key_, [
+             "job_id", "function_id", "function_name", "function", "module",
+             "max_calls"
+         ])
+    assert f._function_descriptor.function_id == ray.FunctionID(
+        function_id_str)
```

Comment:
Could you please add a comment explaining what this is testing?  

It looks like we're only using `function_id_str` here, is there a reason we're requesting all those other fields?  Is it to make sure they were set correctly?